File: backend/photoapp/src/main/java/com/sulay/photoapp/entity/Album.java
package com.sulay.photoapp.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "album")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Album {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @OneToMany(mappedBy = "album", cascade = CascadeType.ALL)
    private List<Photo> photos;

    public void addPhoto(Photo photo) {
        if (this.photos.isEmpty()) {
            this.photos = new ArrayList<>();
        }
        this.photos.add(photo);
    }

}

File: frontend/photo-app/src/app/common/album.ts
export class Album {
    id: number;
    thumbnailUrl: string;
    name: string;
    description: string;

    constructor(id?: number, name?: string, description?: string, thumbnailUrl?: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.thumbnailUrl = thumbnailUrl;
    }

}
